logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/e2ei/usecase/EnrollE2EIUseCase.kt

```diff
@@ -77,18 +78,24 @@ class EnrollE2EIUseCaseImpl internal constructor(
 
         prevNonce = newOrderResponse.second
 
-        val authzResponse = e2EIRepository.createAuthz(prevNonce, newOrderResponse.first.authorizations[0]).getOrFail {
+        val authorizations = e2EIRepository.getAuthorizations(prevNonce, newOrderResponse.first.authorizations).getOrFail {
             return E2EIEnrollmentResult.Failed(E2EIEnrollmentResult.E2EIStep.AcmeNewAuthz, it).toEitherLeft()
         }
 
+        prevNonce = authorizations.second
+        val oidcAuthorizations= authorizations.first[AuthorizationChallengeType.OIDC]!!
+        val dPopAuthorizations = authorizations.first[AuthorizationChallengeType.DPoP]!!
```

Comment:
question: 100% sure we will always have both challenges? If so, why make it a map and deal with nullability? Might as well return something like:

```kotlin
data class ChallengeResult(
    val oicpResult: NewAcmeAuthz,
    val dPopResult: NewAcmeAuthz
)
```